modules/37_industry/subsectors/equations.gms

```diff
@@ -21,5 +24,103 @@ q37_limit_secondary_steel_share(ttot,regi)$( ttot.val ge cm_startyear ) ..
   vm_cesIO(ttot,regi,"ue_steel_secondary")
 ;
 
+*' Compute gross industry emissions before CCS by multiplying sub-sector energy
+*' use with fuel-specific emission factors.
+q37_macBaseInd(ttot,regi,entyFE,secInd37)$( ttot.val ge cm_startyear ) .. 
+  vm_macBaseInd(ttot,regi,entyFE,secInd37)
+  =e=
+    sum(fe2ppfen(entyFE,ppfen_industry_dyn37(in)),
+      vm_cesIO(ttot,regi,in)
+    * p37_fctEmi(entyFE)
+    )
+;
+
+*' Compute maximum possible CCS level in industry sub-sectors given the current
+*' CO2 price.
+q37_emiIndCCSmax(ttot,regi,emiInd37)$( ttot.val ge cm_startyear ) .. 
+  v37_emiIndCCSmax(ttot,regi,emiInd37)
+  =e=
+    !! map sub-sector emissions to sub-sector MACs
+    !! otherInd has no CCS, therefore no MAC, cement has both fuel and process
+    !! emissions under the same MAC
+    sum(emiMac2mac(emiInd37,macInd37),
+      !! add cement process emissions, which are calculated in core/preloop
+      !! from a econometric fit and might not correspond to energy use (FIXME)
+      ( sum((secInd37_2_emiINd37(secInd37,emiInd37),entyFE),
+          vm_macBaseInd(ttot,regi,entyFE,secInd37)
+        )$( NOT sameas(emiInd37,"co2cement_process") )
+      + ( vm_macBaseInd(ttot,regi,"co2cement_process","cement")
+        )$( sameas(emiInd37,"co2cement_process") )
+      )
+    * pm_macSwitch(macInd37)              !! sub-sector CCS available or not
+    * pm_macAbatLev(ttot,regi,macInd37)   !! abatement level at current price
+  )
+;
+
+*' Limit industry CCS to maximum possible CCS level.
+q37_IndCCS(ttot,regi,emiInd37)$( ttot.val ge cm_startyear ) .. 
+  vm_emiIndCCS(ttot,regi,emiInd37)
+  =l=
+  v37_emiIndCCSmax(ttot,regi,emiInd37)
+;
+
+*' Fix cement fuel and cement process emissions to the same abatement level.
+q37_cementCCS(ttot,regi)$(    ttot.val ge cm_startyear
+                          AND pm_macswitch("co2cement")
+                          AND pm_macAbatLev(ttot,regi,"co2cement") ) .. 
+    vm_emiIndCCS(ttot,regi,"co2cement")
+  * v37_emiIndCCSmax(ttot,regi,"co2cement_process")
+  =e=
+    vm_emiIndCCS(ttot,regi,"co2cement_process")
+  * v37_emiIndCCSmax(ttot,regi,"co2cement")
+;
+
+*' Calculate industry CCS costs.
+q37_IndCCSCost(ttot,regi,emiInd37)$( ttot.val ge cm_startyear ) .. 
+  vm_IndCCSCost(ttot,regi,emiInd37)
+  =e=
+    1e-3
+  * pm_macSwitch(emiInd37)
+  * ( sum((enty,secInd37_2_emiInd37(secInd37,emiInd37)),
+        vm_macBaseInd(ttot,regi,enty,secInd37)
+      )$( NOT sameas(emiInd37,"co2cement_process") )
+    + ( vm_macBaseInd(ttot,regi,"co2cement_process","cement")
+      )$( sameas(emiInd37,"co2cement_process") )
+    )
+  * sm_dmac
+  * sum(emiMac2mac(emiInd37,enty),
+      ( pm_macStep(ttot,regi,enty)
+      * sum(steps$( ord(steps) eq pm_macStep(ttot,regi,enty) ),
+          pm_macAbat(ttot,regi,enty,steps)
+        )
+      )
+    - sum(steps$( ord(steps) le pm_macStep(ttot,regi,enty) ),
+        pm_macAbat(ttot,regi,enty,steps)
+      )
+    )
+;
+
+*' FIXME testing
```

Comment:
idem

Reply:
That's part of the special sauce for making secondary steel work.